core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/RequestBody.java

```diff
@@ -47,13 +48,15 @@
     // TODO Handle stream management (progress listener, orig input stream tracking, etc
     private final ContentStreamProvider contentStreamProvider;
     private final long contentLength;
+    private final Optional<Long> optionalContentLength;
     private final String contentType;
 
-    private RequestBody(ContentStreamProvider contentStreamProvider, long contentLength, String contentType) {
+    private RequestBody(ContentStreamProvider contentStreamProvider, Long contentLength, String contentType) {
         this.contentStreamProvider = paramNotNull(contentStreamProvider, "contentStreamProvider");
-        this.contentLength = contentLength;
+        this.contentLength = contentLength != null ? contentLength : 0;
```

Comment:
I'm still not sure if setting the null contentLength as 0 is a good idea, but that's the only way I can think of without breaking change.

Reply:
Since we require content length to be non negative, we can probably use a negative value to denote that content length is absent.

Reply:
Sounds good, will do the change.